pkg/utils/utils.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Nordix Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Go version 1.10 or greater is required. Before that, switching namespaces in
+// long running processes in go did not work in a reliable way.
+// +build go1.10
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	// DefaultCNIDir used for caching hostIFName
+	DefaultCNIDir = "/var/lib/cni/ovs"
```

Comment:
I checked my development cluster and there the CNI directory was still /opt/cni (possibly due to outdated config). I see that we create this directory if not found, so it should work there as well. However, could we make this configurable through the config file?

Also, should this be under /tmp? With /var, if somebody force reboots the node, we will be stuck with leftover configs despite pods they refer to don't exist anymore

Reply:
I think /tmp should be good option, even after node reboot the same cache entry created if same pod created. so shall we use /tmp as default directory and can also override it via config file ?

Reply:
making cache directory as configurable parameter would need to parse args.StdinData in CmdDel to get the directory path, so let's just use /tmp directory now as it satisfies the needs. is the okay ?

Reply:
Sounds good, thanks

Reply:
done